### DISCUSSION TITLE
signal-cli: error: unrecognized arguments: '--trust-new-identities'

### DISCUSSION THREAD

**Discussion by 11notes** [AsamK/signal-cli#857] [Q&A] [answered]
I use Dbus to in integrate signal into my solution, works great, but I have an issue with trust and keys. Every now and then a user will have a new phone and the number needs to be verified, since the daemon can’t do that (no interface present, is one coming? there is none in https://github.com/AsamK/signal-cli/blob/master/man/signal-cli-dbus.5.adoc) I wanted to use the new parameter “--trust-new-identities” and as you can see, it does not recognize it. 

Environment:
```
- Docker (node:14-bullseye)
- signal-cli 0.10.0
```

Start:
```
signal-cli –config /api/etc/signal daemon –system --trust-new-identities always
```

Error:
```
usage: signal-cli daemon [-h] [--dbus] [--dbus-system] [--socket [SOCKET]]\n' +
    '                  [--tcp [TCP]] [--no-receive-stdout]\n' +
    '                  [--receive-mode {on-start,on-connection,manual}]\n' +
    '                  [--ignore-attachments]\n' +
    "signal-cli: error: unrecognized arguments: '--trust-new-identities'\n
```

**Comment by exquo** [comment, marked as answer]
Need to move `--trust-new-identities` before `daemon`. See [FAQ](https://github.com/AsamK/signal-cli/wiki/FAQ#error-parsing-arguments):

	signal-cli  --trust-new-identities always  daemon

Other flags are off too: `--config` is a long option (with _two_ dashes in front); for short, can use `-c`. `-system` should be `--dbus-system`.

**Reply by 11notes** [reply]
Thank you for the quick reply! I’m sorry I did not see that the order of parameters matters. The – was from an autoformat, I use this now and it works. It worked before too, just not with the keys.
```
this.signal_cli = spawn('signal-cli', ['--trust-new-identities', 'always', '-c', '/api/etc/signal', 'daemon', '--dbus-system']);
```

**Reply by 11notes** [reply]
Could I still bother to ask to integrate the key/identity part into the interface on Dbus? Would be easier to run without "--trust-new-identities". That way we could get an event for a new untrusted identity and manually trust and verify it. Like:

```
event: IdentityTrustChanged
Params: Phonenumber, Fingerprint, Safety Number

interface: listuntrusted
interface: listidentities, params: [string:number]
interface: trustidentity, params: string:number, [bool:verify?, string:safety number]
```

With such an event we could send ourselves an info that an end user has a new safety number. What do you think? Possible?

Event: IdentityTrustChanged -> send message to ourselves about the info who has no more trust (phone number, new safety number) -> manually initiate new trust relationship (verify) -> Dbus.trustidentity()

I mean how great of a customer service would that be if you get a call from a company asking you if you changed your phone recently and verifying your new identity via signal. Right now this is not possible without stopping the dbus and running the cli to get the list of untrusted identities, like this we would have an event which is way more dynamic.

**Reply by exquo** [reply]
Yes, I think this is a matter of #826
